File: trees/typescript/Arbol3.ts

type ArbolType<T> = Arbol3<T>
type ArbolMainType<T> = ArbolType<T> | null


export class Arbol3<T> {
  private _key: T
  private _left: ArbolMainType<T>
  private _right1: ArbolMainType<T>
  private _right2: ArbolMainType<T>

  constructor(_key: T, _left: ArbolMainType<T>, _right1: ArbolMainType<T>, _right2: ArbolMainType<T>) {
    this._key = _key
    this._left = _left
    this._right1 = _right1
    this._right2 = _right2
  }

  get key(): T {
    return this._key
  }

  set key(_key: T) {
    this._key = _key
  }

  get left(): ArbolMainType<T> {
    return this._left
  }

  set left(_left: ArbolMainType<T>) {
    this._left = _left
  }

  get right1(): ArbolMainType<T> {
    return this._right1
  }

  set right1(_right1: ArbolMainType<T>) {
    this._right1 = _right1
  }

  get right2(): ArbolMainType<T> {
    return this._right2
  }

  set right2(_right2: ArbolMainType<T>) {
    this._right2 = _right2
  }
}

File: trees/java/Arbol3.java

public class Arbol3 {
  private String key;
  private Arbol3 left;
  private Arbol3 right1;
  private Arbol3 right2;

  public Arbol3(String key, Arbol3 left, Arbol3 right1, Arbol3 right2) {
    this.key = key;
    this.left = left;
    this.right1 = right1;
    this.right2 = right2;
  }

  public String getKey() {
    return this.key;
  }

  public void setKey(String key) {
    this.key = key;
  }

  public Arbol3 getLeft() {
    return this.left;
  }

  public void setLeft(Arbol3 left) {
    this.left = left;
  }

  public Arbol3 getRight1() {
    return this.right1;
  }

  public void setRight1(Arbol3 right1) {
    this.right1 = right1;
  }
  
  public Arbol3 getRight2() {
    return this.right2;
  }

  public void setRight2(Arbol3 right2) {
    this.right2 = right2;
  }
}
